python/pyspark/sql/column.py

```diff
@@ -329,6 +329,31 @@ def getField(self, name):
                 DeprecationWarning)
         return self[name]
 
+    @since(3.1)
+    def withField(self, fieldName, col):
```

Comment:
You should add it into https://github.com/apache/spark/blob/master/python/docs/source/reference/pyspark.sql.rst#dataframe-apis

Reply:
This is part of the Column class not DataFrame class, and I don't see a section for Column yet. Should I add one?

Reply:
Ohhh, okay no need to add in this PR. Let's add it in a separate PR.